spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/bedrock/converse/BedrockConverseProxyChatProperties.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.autoconfigure.bedrock.converse;
+
+import java.time.Duration;
+
+import org.springframework.ai.bedrock.converse.BedrockProxyChatModel;
+import org.springframework.ai.model.function.FunctionCallingOptionsBuilder.PortableFunctionCallingOptions;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.NestedConfigurationProperty;
+import org.springframework.util.Assert;
+
+/**
+ * Configuration properties for Bedrock Converse.
+ *
+ * @author Christian Tzolov
+ * @since 1.0.0
+ */
+@ConfigurationProperties(BedrockConverseProxyChatProperties.CONFIG_PREFIX)
+public class BedrockConverseProxyChatProperties {
+
+	public static final String CONFIG_PREFIX = "spring.ai.bedrock.converse.chat";
+
+	/**
+	 * Set model timeout, Defaults 10 min.
+	 */
+	private Duration timeout = Duration.ofMinutes(5L);
+
+	/**
+	 * Enable Bedrock Converse chat model.
+	 */
+	private boolean enabled = true;
+
+	/**
+	 * The generative id to use. See the {@link BedrockProxyChatModel} for the supported
+	 * models.
+	 */
+	private String model = "anthropic.claude-3-5-sonnet-20240620-v1:0";
+
+	@NestedConfigurationProperty
+	private PortableFunctionCallingOptions options = PortableFunctionCallingOptions.builder()
```

Comment:
With the Bedrock Converse API, different models have different options such as `stop sequences` etc. I wonder if we should add something to set default options for models.

Maybe we could add a new class `BedrockConversePortableFunctionCallingOptions` with a key-value map dynamic field and convert them to Bedrock Converse API's additionalModelRequestFields?

Reply:
Currently the PortableChatOptions and the PortableFunctionCallingOptions allow setting stopSequences and common chat and tool calling options that should work with all models. 

The BedrockConversePortableFunctionCallingOptions with a key-value map dynamic field could be easiest, dynamic,  approach to allow adding additionalModelRequestFields. 
We can add it a follow up PR. 

And in the future we can still add a type-safe options (e.g. dedicated per-model option class).

Reply:
Also we should explore if we can :
- implement additional metrics handler for the Bedrock Converse metrics.
- provide support for the guard features.
- review the cumbersome  streaming tools calling impl.